xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -1266,6 +1292,12 @@ bool CDecoder::Open(AVCodecContext* avctx, AVCodecContext* mainctx, enum AVPixel
   case AV_CODEC_ID_VP9:
     m_refs += 8;
     break;
+  case AV_CODEC_ID_AV1:
+    // the AV1 DXVA2 spec asks for 128 pixel aligned surfaces
+    m_surface_alignment = 128;
+    // by specification AV1 decoder can hold up to 8 unique refs
+    m_refs += 8;
```

Comment:
any reason you don't check `avctx->refs` here like in other codecs?

Reply:
Is because FFmpeg not reports encoding refs since H.264. For example in HEVC always reports 1 refs which is not true. 

I understand that with the current and more complex codecs it is not possible to determine the number of references that the encode has if it is not specified with metadata in the header. Therefore, it is necessary to start decoding to have that information.